graph-builder/src/status.rs

```diff
@@ -64,3 +64,54 @@ pub fn serve_readiness(
     };
     Box::new(future::ok(resp))
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::graph::State;
+    use crate::status;
+    use actix_web::test::TestRequest;
+    use failure::Fallible;
+    use parking_lot::RwLock;
+    use std::collections::HashSet;
+    use std::sync::Arc;
+    use tokio::runtime::current_thread::Runtime;
+
+    fn common_init() -> Fallible<Runtime> {
+        let _ = env_logger::try_init_from_env(env_logger::Env::default());
+        let test_gauge = prometheus::Gauge::new("test_gauge", "test help")?;
+        test_gauge.set(42.0);
+        status::PROM_REGISTRY.register(Box::new(test_gauge))?;
+        Runtime::new().map_err(failure::Error::from)
+    }
+
+    fn mock_state() -> State {
+        let json_graph = Arc::new(RwLock::new(String::new()));
+        let live = Arc::new(RwLock::new(false));
+        let ready = Arc::new(RwLock::new(false));
+
+        State::new(
+            json_graph.clone(),
+            HashSet::new(),
+            live.clone(),
+            ready.clone(),
+        )
+    }
+
+    #[test]
+    fn serve_metrics_basic() -> Fallible<()> {
+        let mut rt = common_init()?;
+
+        let http_req = TestRequest::with_state(mock_state()).finish();
+        let metrics_call = status::serve_metrics(http_req);
+        let resp = rt.block_on(metrics_call)?;
+
+        assert_eq!(resp.status().as_u16(), 200);
+        assert!(resp.body().is_binary());
+
+        if let actix_web::Body::Binary(body) = resp.body() {
+            assert!(!body.is_empty());
```

Comment:
Since you're using `failure` in here you can use the `ensure!()` macro here and elsewhere instead of `assert!()`

Reply:
`assert` and `ensure` are not directly interchangeable, as the latter doesn't have fileline information and additional introspection details provided by the former.

I'm sticking to ensure for enforcing invariants, but feel free to have a spike ticket to investigate/build a blended `ensure`+`assert_*` macro.